http.c

```diff
@@ -3864,6 +3873,25 @@ evhttp_set_allowed_methods(struct evhttp* http, ev_uint16_t methods)
 	http->allowed_methods = methods;
 }
 
+void
+evhttp_set_max_connection_ttl(struct evhttp* http, int max_ttl)
+{
+	if (max_ttl <= 0) {
+		http->connection_max_ttl = 0;
+
+		if (event_pending(&http->expire_event, EV_TIMEOUT, 0))
+			event_del(&http->expire_event);
+
+	} else {
+		http->connection_max_ttl = max_ttl;
+
+		if (!event_pending(&http->expire_event, EV_TIMEOUT, 0)) {
+			struct timeval timeout = {60, 0}; /* check once per minute */
```

Comment:
It max_ttl expressed in seconds, how does expire_event timer could be once per minute?

Reply:
That's a good question.  I could change the max TTL unit to minutes if you like.  The reason for checking only once per minute is that this operation is ``O(n)`` where ``n`` is the number of connections.  I wanted to keep it reasonable.  I don't really like having a magic number (i.e. 60 seconds) hidden in the code but making this configurable would seem to unnecessarily complicate the API.  What do you think?

Reply:
Well, maybe this requires more code in this case, for example priority queue.

Another possibility is to use timer event per connection/request (I remember that you want this functionality for connections but it may be reasonable for requests to limit their time not on read/write/connect basis), and just add ability to add it per-connection/http-server-base (like evhttp_set_timeout)

Reply:
To mitigate slowloris attacks it really has to be per connection.  Otherwise, an attacker could simply end each request within the timeout and start a new one.

A per connection timer event would work well.